go/logic/applier.go

```diff
@@ -73,7 +73,7 @@ func (this *Applier) InitDBConnections() (err error) {
 	if this.db, _, err = mysql.GetDB(this.migrationContext.Uuid, applierUri); err != nil {
 		return err
 	}
-	singletonApplierUri := fmt.Sprintf("%s?timeout=0", applierUri)
+	singletonApplierUri := fmt.Sprintf("%s&timeout=0", applierUri)
```

Comment:
This was an existing bug we found when adding the tls option onto the uri. Previously, this was causing the uri to take the form of `gromit:penguin@tcp(myhost:3306)/test?interpolateParams=true&autocommit=true&charset=utf8mb4,utf8,latin1?timeout=0`. The DSN parser for go-mysql-driver/mysql was then not parsing the timeout setting, but instead including that as part of the value for `charset`.

Reply:
What if `applierUri` has no query string yet?

Reply:
@johnlockwood-wf `applierUri` has query strings from the return value of [ConnectionConfig.GetDBUri](https://github.com/github/gh-ost/pull/705/files#diff-0ec6a3deacd5e2577724a82d8347226fR101) both before and after this PR.

Reply:
I rather not refactor every place where GetDBUri is currently being used to address this bug. Because we are adding the tls param into GetDBUri, we are directly impacted by this line using the wrong symbol and thus I am including the fix here to use the correct symbol.

Additionally, these are not really URI's but DSNs recognized by [go-my-sql/driver](https://github.com/go-sql-driver/mysql#dsn-data-source-name). While [net/url.Values](https://golang.org/pkg/net/url/#Values) may have some helpful functions, I don't think the work involved in shoehorning that in is worthwhile here where each code path that hits this spot already has params set for `applierUri`.

Reply:
go-sql-driver's [Config.FormatDSN](https://godoc.org/github.com/go-sql-driver/mysql#Config.FormatDSN) would likely be a great improvement on the manually created DSN, but I think that is something to reserve for a separate issue or PR.

Reply:
Agreed. Let's postpone use of `FormatDNS` to a next PR.